File: src/main/java/backend/controller/RequestBodies/Post.java
package backend.controller.RequestBodies;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.Date;

public class Post {

    @NotNull
    private Date created;

    @NotNull
    private Date sent;

    @NotNull
    private String text;

    @NotNull
    @Valid
    private Location location;

    @NotNull
    private String challengeId;

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public Date getSent() {
        return sent;
    }

    public void setSent(Date sent) {
        this.sent = sent;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public String getChallengeId() {
        return challengeId;
    }

    @JsonProperty("challenge_id")
    public void setChallengeId(String challengeId) {
        this.challengeId = challengeId;
    }
}

File: src/main/java/backend/controller/RequestBodies/Post.kt
@file:JvmName("Post")
package backend.controller.RequestBodies

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*
import javax.validation.Valid
import javax.validation.constraints.NotNull

class Post {
    @NotNull var created: Date? = null
    @NotNull var sent: Date? = null
    @NotNull var text: String? = null
    @NotNull @Valid var location: Location? = null
    @NotNull @JsonProperty("challenge_id") var challengeId: String? = null
}
